Contrib/FreeWilson/freewilson.py

```diff
@@ -0,0 +1,486 @@
+                                    default_decomp_params = rdkit.Chem.rdRGroupDecomposition.RGroupDecompositionParameters()
+                                    default_decomp_params.matchingStrategy = rgd.GA
+                                    default_decomp_params.onlyMatchAtRGroups = False
+                                   ]
+                                If you only want to decompose on specific group locations
+                                set onlyMatchAtRGroups to True
+
+
+        >>> from rdkit import Chem
+        >>> from rdkit.Chem import Descriptors, FreeWilson
+        >>> scaffold = Chem.MolFromSmiles("c1cccnc1")
+        >>> mols = [Chem.MolFromSmiles("c1cccnc1"+"C"*(i+1)) for i in range(100)]
+        >>> scores = [Descriptors.MolLogP(m) for m in mols]
+        >>> fw = FWDecompose(scaffold, mols, scores)
+        >>> for pred in FWBuild(fw):
+        >>>  ...
+
+    To filter out molecules in the training set, send the freewilson decomoposition into
+    the builer.
+
+       >>> for pred in FWBuild(fw, decomposition=fw):
+       >>>    ...
+
+    For an easy way to report predictions see 
+
+       >>> import sys
+       >>> predictions_to_csv(sys.stdout, fw, FWBuild(fw))
+
+   
+    See FWBuild docs to see how to filter predictions, molecular weight or molecular properties.
+    """
+    descriptors = descriptors = [] # list of descriptors, one per matched molecules
+                                   #    descriptors are 1/0 if a sidechain is present
+    matched_scores = []            # scores from the matching molecules
+    rgroup_idx = rgroup_idx = {}   # rgroup index into descriptor { smiles: idx }
+    rgroups = rgroups = defaultdict(list) # final list of rgrups/sidechains
+
+    if len(mols) != len(scores):
+        raise ValueError(f"The number of molecules must match the number of scores #mols {len(mols)} #scores {len(scores)}")
+    # decompose the rgroups
+    logger.info(f"Decomposing {len(mols)} molecules...")
+    decomposer = rgd.RGroupDecomposition(scaffolds, decomp_params)
+    for mol, score in tqdm(zip(mols, scores)):
+        if decomposer.Add(mol) >= 0:
+            matched_scores.append(score)
+    decomposer.Process()
+    logger.info(f"Matched {len(matched_scores)} out of {len(mols)}")
+    if not(matched_scores):
+        logger.error("No scaffolds matched the input molecules")
+        return
+
+    decomposition = decomposition = decomposer.GetRGroupsAsRows(asSmiles=True)
+    logger.info("Get unique rgroups...")
+    rgroup_counts = defaultdict(int)
+    for row in decomposition:
+        for rgroup,smiles in row.items():
+            rgroup_counts[smiles] += 1
+            if smiles not in rgroup_idx:
+                rgroup_idx[smiles] = len(rgroup_idx)
+                rgroups[rgroup].append(RGroup(smiles, rgroup, 0, 0))
+
+    logger.info(f"Descriptor size {len(rgroup_idx)}")
+    # get the descriptors list, one-hot encoding per rgroup
+    for i,row in enumerate(decomposition):
```

Comment:
you don't use i, so:
```suggestion
    for row in decomposition:
```